scene/gui/label.cpp

```diff
@@ -288,6 +288,30 @@ void Label::_update_theme_item_cache() {
 	theme_cache.font_shadow_outline_size = get_theme_constant(SNAME("shadow_outline_size"));
 }
 
+PackedStringArray Label::get_configuration_warnings() const {
+	PackedStringArray warnings = Control::get_configuration_warnings();
+
+	Ref<Font> font;
+	if (settings.is_valid()) {
+		font = settings->get_font();
+	}
+	if (font.is_null()) {
+		font = theme_cache.font;
+	}
+
+	String txt = (uppercase) ? TS->string_to_upper(xl_text, language) : xl_text;
+	if (font.is_valid()) {
+		for (int i = 0; i < txt.length(); i++) {
+			if (!font->has_char(txt.unicode_at(i))) {
+				warnings.push_back(RTR("The current font does not support rendering one or more characters used in this Label's text."));
+				break;
+			}
+		}
+	}
```

Comment:
A more efficient and correct way is to check shaped text data for glyphs with index 0 and empty font rid instead.
Char is not necessary equal to the displayed glyph, and it's possible for a font to have a base char form but not a context specific one.

Reply:
Thanks for the hint! Updated the code and it still works as expected (with my limited testing of emojis) :)